event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/AstyanaxEventWriterDAO.java

```diff
@@ -62,9 +62,17 @@ public void addAll(Multimap<String, ByteBuffer> eventsByChannel, @Nullable Event
                 Duration eventTtl = _channelConfiguration.getEventTtl(channel);
 
                 int remaining = events.size();
+                long remainingSizeInBytes = events.stream().mapToLong(event->event.array().length).sum();
+                long numOversizeEvents = events.stream().filter(event->event.array().length > Constants.MAX_SLAB_SIZE_IN_BYTES).count();
+
+                // If there are oversize events slab allocation will fail so throw an exception
+                if (numOversizeEvents > 0) {
+                    throw new RuntimeException("AstyanaxEventWriterDAO.addAll: events (documents) present that are larger than max size of " + Constants.MAX_SLAB_SIZE_IN_MEGABYTES + " MB");
+                }
+
                 while (remaining > 0) {
                     // Each allocation will be from a different slab and hence a different row.
-                    SlabAllocation allocation = _slabAllocator.allocate(channel, remaining);
+                    SlabAllocation allocation = _slabAllocator.allocate(channel, remaining, remainingSizeInBytes);
```

Comment:
Doesn't remainingSizeInBytes change after each allocation? It seems we are using a static value calculated above.

Reply:
yes it does!